app/src/main/java/space/kis/testproject/fragments/RetrofitFragment.kt

```diff
@@ -0,0 +1,122 @@
+package space.kis.testproject.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.appcompat.widget.SearchView
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import space.kis.domain.model.LceState
+import space.kis.testproject.R
+import space.kis.testproject.adapter.CountryListAdapter
+import space.kis.testproject.databinding.FragmentRetrofitBinding
+import space.kis.testproject.viewmodels.RetrofitViewModel
+
+class RetrofitFragment : Fragment() {
+
+    private var _binding: FragmentRetrofitBinding? = null
+    private val binding get() = requireNotNull(_binding)
+
+    private val viewModel by viewModel<RetrofitViewModel>()
+
+    private val adapter by lazy {
+        CountryListAdapter(itemClick = {
+            findNavController().navigate(
+                RetrofitFragmentDirections.actionListFromRetrofitFragmentToDetailsFragment(
+                    it.name, it.capital, it.area.toString(), it.population.toString(), it.flag
+                )
+            )
+        }, longItemClick = {
+            viewLifecycleOwner.lifecycleScope.launch(Dispatchers.IO) {
+                viewModel.addCountryToDb(
+                    it.id,
+                    it.name,
+                    it.capital,
+                    it.area,
+                    it.population,
+                    it.flag
+                )
+            }
```

Comment:
здесь должно быть просто 
```suggestion
            viewModel.onCountryLongClicked(it)
```
сама вью модель решает, что делать с этим кликом